src/mlang/backend_ir/bir_interface.ml

```diff
@@ -201,7 +233,17 @@ let adapt_program_to_function (p : Bir.program) (f : bir_function) : Bir.program
                     {
                       Mir.var_typ = None;
                       Mir.var_io = Regular;
-                      Mir.var_definition = Mir.SimpleVar (Mir.Literal Mir.Undefined, pos);
+                      Mir.var_definition =
```

Comment:
Good catch!

Reply:
All the praise goes to Keryan for this one!